tests/hlsl-intrinsic/shader-execution-reordering/hit-object-trace-motion-ray.slang.1.expected

```diff
@@ -0,0 +1,96 @@
+result code = 0
+standard error = {
+}
+standard output = {
+#version 460
+#extension GL_EXT_ray_tracing : require
+#extension GLSL_NV_ray_tracing_motion_blur : require
+#extension GL_NV_shader_invocation_reorder : require
+layout(row_major) uniform;
+layout(row_major) buffer;
+struct SomeValues_0
+{
+    int a_0;
+    float b_0;
+};
+
+layout(location = 0)
+rayPayloadEXT
+SomeValues_0 p_0;
+
+struct RayDesc_0
+{
+    vec3 Origin_0;
+    float TMin_0;
+    vec3 Direction_0;
+    float TMax_0;
+};
+
+hitObjectNV HitObject_TraceMotionRay_0(accelerationStructureEXT AccelerationStructure_0, uint RayFlags_0, uint InstanceInclusionMask_0, uint RayContributionToHitGroupIndex_0, uint MultiplierForGeometryContributionToHitGroupIndex_0, uint MissShaderIndex_0, RayDesc_0 Ray_0, float CurrentTime_0, inout SomeValues_0 Payload_0)
+{
+    p_0 = Payload_0;
+    hitObjectNV hitObj_0;
+    hitObjectTraceRayMotionNV(hitObj_0, AccelerationStructure_0, RayFlags_0, InstanceInclusionMask_0, RayContributionToHitGroupIndex_0, MultiplierForGeometryContributionToHitGroupIndex_0, MissShaderIndex_0, Ray_0.Origin_0, Ray_0.TMin_0, Ray_0.Direction_0, Ray_0.TMax_0, CurrentTime_0, (0));
+    Payload_0 = p_0;
+    return hitObj_0;
+}
+
+layout(binding = 0)
+uniform accelerationStructureEXT scene_0;
+
+layout(location = 0)
+hitObjectAttributeNV
+SomeValues_0 t_0;
+
+SomeValues_0 HitObject_GetAttributes_0(hitObjectNV this_0)
+{
+    return t_0;
+}
+
+uint calcValue_0(hitObjectNV hit_0)
+{
+    bool _S1 = (hitObjectIsHitNV((hit_0)));
+    uint r_0;
+    if(_S1)
+    {
+        uint instanceIndex_0 = (hitObjectGetInstanceCustomIndexNV((hit_0)));
+        uint instanceID_0 = (hitObjectGetInstanceIdNV((hit_0)));
+        uint geometryIndex_0 = (hitObjectGetGeometryIndexNV((hit_0)));
+        uint primitiveIndex_0 = (hitObjectGetPrimitiveIndexNV((hit_0)));
+        SomeValues_0 objSomeValues_0 = HitObject_GetAttributes_0(hit_0);
+        r_0 = 0U + instanceIndex_0 + instanceID_0 + geometryIndex_0 + primitiveIndex_0 + uint(objSomeValues_0.a_0);
+    }
+    else
+    {
+        r_0 = 0U;
+    }
+    return r_0;
+}
+
+layout(std430, binding = 1) buffer _S2 {
+    uint _data[];
+} outputBuffer_0;
+void main()
+{
+    uvec3 _S3 = ((gl_LaunchIDEXT));
+    ivec2 launchID_0 = ivec2(_S3.xy);
+    uvec3 _S4 = ((gl_LaunchSizeEXT));
+    int idx_0 = launchID_0.x;
+    int _S5 = idx_0 / 4;
+    float currentTime_0 = float(_S5);
+    SomeValues_0 someValues_0;
+    SomeValues_0 _S6 = { idx_0, float(idx_0) * 2.00000000000000000000 };
+    someValues_0 = _S6;
+    RayDesc_0 ray_0;
+    ray_0.Origin_0 = vec3(float(idx_0), 0.00000000000000000000, 0.00000000000000000000);
+    ray_0.TMin_0 = 0.00999999977648258209;
+    ray_0.Direction_0 = vec3(0.00000000000000000000, 1.00000000000000000000, 0.00000000000000000000);
+    ray_0.TMax_0 = 10000.00000000000000000000;
+    hitObjectNV hit_1 = HitObject_TraceMotionRay_0(scene_0, 20U, 255U, 0U, 4U, 0U, ray_0, currentTime_0, someValues_0);
```

Comment:
Generated GLSL is illegal.
`hitObjectNV` is an opaque type, you cannot assign/initialize it.

This is exactly similar in spirit to `rayQueryEXT` (with all its syntactic limitations)
So whatever codegen slang does for translating `RayQuery`, same should be done for `hitObjectNV`

Reply:
Ok I see. That makes some parts of the API make more sense, as it didn't seem to map similarly to the NVAPI version of things. Will look to make suitable changes.

Reply:
@alelenv There remain some issues, but it looks like the output for these tests appear appropriate. Would you mind taking a quick look?